pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -445,6 +449,13 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 			return reconcile.Result{}, err
 		}
 
+		jobHash, err := calculateJobSpecHash(job)
+		if err != nil {
+			cdLog.WithError(err).Error("failed to calulcate hash for generated install job")
+			return reconcile.Result{}, err
+		}
+		job.Annotations[jobHashAnnotation] = jobHash
```

Comment:
i was going to do something like that, but then i saw that GenerateInstallJob() always adds its own annotations https://github.com/openshift/hive/blob/master/pkg/install/generate.go#L82

i can add it for extra safety...WDYT?

Reply:
Yeah it would be good to add for extra safety. If we ever change what we generate and remove annotations we may not know that it'll affect this part of the code.

Reply:
addressed with most recent commit